### PR TITLE
feat: Obsolete product handling

### PR DESCRIPTION
**Pull request by john-gom**

### CODE REVIEW

**Review by alexgarel — APPROVED** [review]
Good !

File: src/domain/services/query.service.ts [hunk 1]
```diff
@@ -85,16 +86,23 @@ export class QueryService {
...
+    const obsolete = !!body?.obsolete;
+    delete body?.obsolete;
+    return `${obsolete ? '' : 'not '}pt.obsolete`;
+  }
+
   async count(body: any) {
     const start = Date.now();
     this.logger.debug(body);
 
+    const obseleteWhere = this.obsoleteWhere(body);
```
Comment (stephanegigandet) [inline src/domain/services/query.service.ts:99]:
```suggestion
    const obsoleteWhere = this.obsoleteWhere(body);
```

Comment (alexgarel) [reply src/domain/services/query.service.ts:99]: eagle eye :wink:

**Review by stephanegigandet — APPROVED** [review]
Thank you!

File: src/domain/services/query.service.ts (hunk 1 above)
Comment (john-gom) [reply src/domain/services/query.service.ts:99]: I really have trouble spelling that word!

**Pull request merged**